src/app/gateways/views/BuiltinGatewayDetailView.vue

```diff
@@ -1,242 +1,45 @@
 <template>
-  <DataSource
-    v-slot="{ data: me }: MeSource"
-    src="/me"
+  <RouteView
+    v-slot="{ route }"
+    name="builtin-gateway-detail-view"
+    :params="{
+      mesh: '',
+      gateway: '',
+      listener: '0',
```

Comment:
I don't know if you saw my comment offline, but I think we can use a `hostname:port` combo for an identifier for these? wdyt?

Reply:
I’m not entirely certain that is reliable. There is certainly validation in place preventing a clash of `${hostname}:${port}`, but I don’t know if there are edge cases that make this more tricky than it seems. While it’s not exactly pretty, I think the index is a reliable way to identify a listener for now (until someone adds an non-last listener, but the same problem exists when changing hostname or port).

Also, note that as far as I can tell, only the port is required. In the case of an absent hostname, the hostname is implied to be `*`.

Slightly related: I would like for the `/_rules` API to identify which listeners their rules apply to. This, too, would need to use _some_ form of identifier (even if it’s just the listener index).

Reply:
Yeah I'm not 100% sure either, but considering there is validation stopping you from adding a clash of a `hostname:port` combo, I'd say its extremely unlikely we can't rely on it. Again considering there is validation preventing you from doing this, I'd almost say it would be a bug if someone did manage to do that.

Absent hostname would produce a `/*:80/` in the URL.

---

Separately, if we are using the URLs as state (ie. bookmarkable, sharable state), what happens if I share a link to listener `/3/` and then I delete listener number 2? Listener 3 would become listener 2 right? Then the link I just shared would be wrong.

Reply:
> what happens if I share a link to listener `/3/` and then I delete listener number 2? Listener 3 would become listener 2 right?

Yes. But the link to whatever `${hostname}:${port}` would’ve been for listener 2 would also no longer work. 🤷🏻 

I’ll consider updating this, but I might do it in a follow up.

Reply:
> But the link to whatever ${hostname}:${port} would’ve been for listener 2 would also no longer work.

And that would be the correct behaviour becuase I've deleted it. "Listener 3" would still be available at the same link of `hostname:port`.

Reply:
For the sake of trying how this would work, I implemented this locally and there’s a problem with it: we don’t know `${hostname}:${port}` ahead of getting the MeshGateway. That means we can’t load the view with a pre-selected listener. We would need to update the URL after loading the MeshGateway. I’m going to leave this as an index for now.